herbert_core/classes/MPIFramework/@iMessagesFramework/private/check_get_persistent_.m

```diff
@@ -0,0 +1,41 @@
+function [mess,id_from] = check_get_persistent_(obj,source_address)
+% check if a message is a persistent message (the message
+% describing a state of the source which persists until the
+% current job is completed or aborted) and return these
+% persistent messages.
+% Input:
+% source_address -- the array of addresses to check for sources
+%                   of the persistent messages
+% Returns:
+% mess   -- cellarray of persisting messages returned from all
+%           or some sources requested
+%id_from -- array of the addresses which have previously
+%           generated persistent messages, stored within the
+%           framework
+
+mess = {};
+id_from = source_address;
+if isempty(obj.persistent_fail_message_)
+    return;
+end
+if isempty(source_address) || (ischar(source_address) && strcmp(source_address,'any'))
+    id_from = obj.persistent_fail_message_.keys();
+    id_from = [id_from{:}];
+    mess = obj.persistent_fail_message_.values();
+    return;
```

Comment:
is this `return` needed? Isn't it just an `elseif`?

Reply:
one can probably do it through elseif, but large branches of if-s are not easy to understand. 
here I am just finish checks in case of special conditions.

Reply:
Nor are sections of code with return statements scattered through them.